src/components/ConnectionsList.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useEffect, useState } from "react";
+import { Button } from "antd";
+import ConnectionsListProfiles from "./ConnectionsListProfiles";
+import { useAppDispatch, useAppSelector } from "../redux/hooks";
+import * as sdk from "../services/sdk";
+import { Graph, HexString, Profile } from "../utilities/types";
+import { upsertProfile } from "../redux/slices/profileSlice";
+
+enum ListStatus {
+  CLOSED,
+  FOLLOWERS,
+  FOLLOWING,
+}
+
+const ConnectionsList = (): JSX.Element => {
+  const profile: Profile | undefined = useAppSelector(
+    (state) => state.user.profile
+  );
+  const graphs: Graph[] = useAppSelector((state) => state.graphs.graphs);
+  const graph: Graph | undefined = graphs.find(
+    ({ socialAddress }) => socialAddress === profile?.socialAddress
+  );
+  const cachedProfiles: Profile[] = useAppSelector(
+    (state) => state.profiles.profiles
+  );
+  const [selectedListTitle, setSelectedListTitle] = useState<ListStatus>(
+    ListStatus.CLOSED
+  );
+  const [selectedList, setSelectedList] = useState<Profile[]>([]);
+  const [followingList, setFollowingList] = useState<Profile[]>([]);
+  const [followersList, setFollowersList] = useState<Profile[]>([]);
+  const [notFollowing, setNotFollowing] = useState<Profile[]>([]);
+  const dispatch = useAppDispatch();
+
+  const getConnectionProfile = async (
```

Comment:
getConnectionProfile for a name is a little misleading because it also upserts a profile.  I think decoupling the getting and upsert will make thing a little more clear.

Reply:
but the upsert is only one line of code and i think it is more readable to have there.  the order of operations is very clear compared to pulling it out into another function.

Reply:
maybe rename to upsertConnectionProfile